kmath-functions/src/commonMain/kotlin/space/kscience/kmath/functions/Polynomial.kt

```diff
@@ -5,128 +5,498 @@
 
-/**
- * Create a polynomial witch represents differentiated version of this polynomial
- */
-@UnstableKMathAPI
-public fun <T, A> Polynomial<T>.differentiate(
-    algebra: A,
-): Polynomial<T> where  A : Ring<T>, A : NumericAlgebra<T> = algebra {
-    Polynomial(coefficients.drop(1).mapIndexed { index, t -> number(index) * t })
-}
+    /**
+     * Returns sum of the polynomial and the integer represented as a polynomial.
+     *
+     * The operation is equivalent to adding [other] copies of unit polynomial to [this].
+     */
+    public operator fun P.plus(other: Int): P = addMultipliedByDoubling(this, one, other)
+    /**
+     * Returns difference between the polynomial and the integer represented as a polynomial.
+     *
+     * The operation is equivalent to subtraction [other] copies of unit polynomial from [this].
+     */
+    public operator fun P.minus(other: Int): P = addMultipliedByDoubling(this, one, -other)
+    /**
+     * Returns product of the polynomial and the integer represented as a polynomial.
+     *
+     * The operation is equivalent to sum of [other] copies of [this].
+     */
+    public operator fun P.times(other: Int): P = multiplyByDoubling(this, other)
 
-/**
- * Create a polynomial witch represents indefinite integral version of this polynomial
- */
-@UnstableKMathAPI
-public fun <T, A> Polynomial<T>.integrate(
-    algebra: A,
-): Polynomial<T> where  A : Field<T>, A : NumericAlgebra<T> = algebra {
-    val integratedCoefficients = buildList(coefficients.size + 1) {
-        add(zero)
-        coefficients.forEachIndexed{ index, t -> add(t / (number(index) + one)) }
-    }
-    Polynomial(integratedCoefficients)
+    /**
+     * Returns sum of the integer represented as a polynomial and the polynomial.
+     *
+     * The operation is equivalent to adding [this] copies of unit polynomial to [other].
+     */
+    public operator fun Int.plus(other: P): P = addMultipliedByDoubling(other, one, this)
+    /**
+     * Returns difference between the integer represented as a polynomial and the polynomial.
+     *
+     * The operation is equivalent to subtraction [this] copies of unit polynomial from [other].
+     */
+    public operator fun Int.minus(other: P): P = addMultipliedByDoubling(-other, one, this)
+    /**
+     * Returns product of the integer represented as a polynomial and the polynomial.
+     *
+     * The operation is equivalent to sum of [this] copies of [other].
+     */
+    public operator fun Int.times(other: P): P = multiplyByDoubling(other, this)
+
+    /**
+     * Converts the integer [value] to polynomial.
+     */
+    public fun number(value: Int): P = number(constantNumber(value))
+    /**
+     * Converts the integer to polynomial.
+     */
+    public fun Int.asPolynomial(): P = number(this)
+
+    /**
+     * Returns the same constant.
+     */
+    @JvmName("unaryPlusConstant")
+    @JsName("unaryPlusConstant")
+    public operator fun C.unaryPlus(): C = this
```

Comment:
I would like to avoid duplicating algebra functionality this way. It is better to use nested algebra scopes.